Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1048,41 +1050,53 @@ - (NSString *)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andTyp
     return shortURL;
 }
 
-- (NSString *) generateLongURLWithParams:(NSDictionary *)params andChannel:(NSString *)channel andTags:(NSArray *)tags andFeature:(NSString *)feature andStage:(NSString *)stage andAlias:(NSString *)alias {
-    NSMutableString *longURL = [BNC_LINK_URL mutableCopy];
+- (NSString *)generateLongURLWithParams:(NSDictionary *)params andChannel:(NSString *)channel andTags:(NSArray *)tags andFeature:(NSString *)feature andStage:(NSString *)stage andAlias:(NSString *)alias {
+    NSString *appIdentifier = [BNCPreferenceHelper getBranchKey];
+    if ([appIdentifier isEqualToString:NO_STRING_VALUE]) {
+        appIdentifier = [BNCPreferenceHelper getAppKey];
+    }
     
-    [longURL appendString:[NSString stringWithFormat:@"/a/%@?", [BNCPreferenceHelper getAppKey]]];
+    if ([appIdentifier isEqualToString:NO_STRING_VALUE]) {
+        NSLog(@"No Branch Key specified, cannot create a long url");
+        return nil;
+    }
     
-    NSData *jsonData = [BNCEncodingUtils encodeDictionaryToJsonData:params];
-    NSString *base64EncodedParams = [BNCEncodingUtils base64EncodeData:jsonData];
+    NSString *baseLongUrl = [NSString stringWithFormat:@"%@/a/%@", BNC_LINK_URL, appIdentifier];
+
+    return [self longUrlWithBaseUrl:baseLongUrl params:params tags:tags feature:feature channel:nil stage:stage alias:alias duration:0 type:BranchLinkTypeUnlimitedUse];
```

Comment:
just curious -- `BranchLinkTypeUnlimitedUse` is the default parameter here right?